File: java/class 14/Card.java

public class Card {

	final static String SPADE = "Spade";
	final static String HEART = "Heart";
	final static String CLUB = "Club";
	final static String DIAMOND = "Diamond";

	final static String[] SUITS = { SPADE, HEART, CLUB, DIAMOND };
	final static String[] RANKS = { "A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K" };

	String suit;
	String rank;
	boolean marked = false;
	
	public Card(String suit, String rank) {
		this.suit = suit;
		this.rank = rank;
	}

	public String getRank() {
		return this.rank;
	}
	
	public String getSuit() {
		return this.suit;
	}

	public String getName() {
		return this.suit + " " + this.rank;
	}
	
	public boolean isMarked() {
		return this.marked;
	}
	
	public void setMarked(boolean marked) {
		this.marked = marked;
	}
}

File: python/class 17-18/Card.py
class Card:
    SPADE = "Spade"
    HEART = "Heart"
    CLUB = "Club"
    DIAMOND = "Diamond"

    Suits = (SPADE, HEART, CLUB, DIAMOND)
    Ranks = ("A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K")

    def __init__(self, suit, rank):
        self.suit = suit
        self.rank = rank
        self.marked = False

    def getRank(self):
        return self.rank

    def getSuit(self):
        return self.suit

    def getName(self):
        return self.suit + " " + self.rank

    def isMarked(self):
        return self.marked

    def setMarked(self, marked = True):
        self.marked = marked

